src/app/protocol-handler.ts

```diff
@@ -112,29 +113,46 @@ class ProtocolHandler {
    * Sets session cookies and navigates to the pod url
    */
   public async handleSeamlessLogin(protocolUri: string): Promise<void> {
-    const { url } = config.getUserConfigFields(['url']);
+    const globalConfig = config.getGlobalConfigFields(['url']);
+    const userConfig = config.getUserConfigFields(['url']);
+    const url = userConfig.url ? userConfig.url : globalConfig.url;
+    const { subdomain, tld, domain } = whitelistHandler.parseDomain(url);
+    const cookieDomain = `.${subdomain}.${domain}${tld}`;
     if (protocolUri) {
       const urlParams = new URLSearchParams(new URL(protocolUri).search);
       const skeyValue = urlParams.get('skey');
       const anticsrfValue = urlParams.get('anticsrf');
-      if (skeyValue) {
-        await session.defaultSession.cookies.set({
-          url,
-          name: 'skey',
-          value: skeyValue,
-        });
-      }
-      if (anticsrfValue) {
-        await session.defaultSession.cookies.set({
-          url,
-          name: 'anti-csrf-cookie',
-          value: anticsrfValue,
-        });
+      if (skeyValue && anticsrfValue) {
+        try {
+          await session.defaultSession.cookies.set({
+            url,
```

Comment:
can we define an interface for this object for reusing purpose and for indication?